### ISSUE TITLE
Cannot parse PostgreSQL full text search operator

### ISSUE THREAD

**Issue by jasonaowen** [sqlfluff/sqlfluff#4837]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

PostgreSQL offers sophisticated [full text search](https://www.postgresql.org/docs/current/textsearch-intro.html) capabilities. It does so via the [text search operator `@@`](https://www.postgresql.org/docs/current/functions-textsearch.html). For example:

```
=> select to_tsvector('examples') @@ websearch_to_tsquery('example');
 ?column? 
----------
 t
(1 row)
```

However, SQLFluff currently cannot parse this query.

### Expected Behaviour

Parsing this file should succeed:

```sql
select to_tsvector('examples') @@ websearch_to_tsquery('example');
```

### Observed Behaviour

```
$ sqlfluff parse test-search.sql 
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'to_tsvector'
[L:  1, P: 19]      |                        bracketed:
[L:  1, P: 19]      |                            start_bracket:                    '('
[L:  1, P: 20]      |                            [META] indent:
[L:  1, P: 20]      |                            expression:
[L:  1, P: 20]      |                                quoted_literal:               "'examples'"
[L:  1, P: 30]      |                            [META] dedent:
[L:  1, P: 30]      |                            end_bracket:                      ')'
[L:  1, P: 31]      |                whitespace:                                   ' '
[L:  1, P: 32]      |                unparsable:                                   !! Expected: 'Nothing...'
[L:  1, P: 32]      |                    raw:                                      '@'
[L:  1, P: 33]      |                    raw:                                      '@'
[L:  1, P: 34]      |                    whitespace:                               ' '
[L:  1, P: 35]      |                    code:                                     'websearch_to_tsquery'
[L:  1, P: 55]      |                    bracketed:
[L:  1, P: 55]      |                        start_bracket:                        '('
[L:  1, P: 56]      |                        single_quote:                         "'example'"
[L:  1, P: 65]      |                        end_bracket:                          ')'
[L:  1, P: 66]      |            [META] dedent:
[L:  1, P: 66]      |    statement_terminator:                                     ';'
[L:  1, P: 67]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:  32 |  PRS | Line 1, Position 32: Found unparsable section: "@@
                       | websearch_to_tsquery('example')"
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
```

### How to reproduce

`test-search.sql`:

```sql
select to_tsvector('examples') @@ websearch_to_tsquery('example');
```

```sh
sqlfluff parse test-search.sql
```

### Dialect

`postgres`

### Version

```
$ sqlfluff --version
sqlfluff, version 2.0.7
$ python --version
Python 3.10.6
```

### Configuration

```ini
[sqlfluff]
dialect = postgres
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by jasonaowen** [comment]
It looks like this was mostly fixed in #6323 - thank you, @keraion! The only full-text search operator left is `!!`:

> `!! tsquery` → `tsquery`
> 
> Negates a `tsquery`, producing a query that matches documents that do not match the input query.
> 
> `!! 'cat'::tsquery` → `!'cat'`

Example query:

```sql
select !!websearch_to_tsquery('example');
```

`sqlfluff parse` result:

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                [META] dedent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                unparsable:                                   !! Expected: 'Nothing here.'
[L:  1, P:  8]      |                    not:                                      '!'
[L:  1, P:  9]      |                    not:                                      '!'
[L:  1, P: 10]      |                    word:                                     'websearch_to_tsquery'
[L:  1, P: 30]      |                    start_bracket:                            '('
[L:  1, P: 31]      |                    single_quote:                             "'example'"
[L:  1, P: 40]      |                    end_bracket:                              ')'
[L:  1, P: 41]      |    statement_terminator:                                     ';'
[L:  1, P: 42]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:   8 |  PRS | Line 1, Position 8: Found unparsable section:
                       | "!!websearch_to_tsquery('example')"
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
```
